sklearn/decomposition/tests/test_nmf.py

```diff
@@ -589,12 +707,66 @@ def test_nmf_custom_init_dtype_error():
     W = rng.random_sample((20, 15))
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
-        NMF(init='custom').fit(X, H=H, W=W)
+        Estimator(init='custom').fit(X, H=H, W=W)
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
         non_negative_factorization(X, H=H, update_H=False)
 
 
+def test_nmf_is_minibatch_nmf():
+    # Test that the standard nmf is the minibatch nmf after 1 iteration
+    # with batch_size = n_samples and forget_factor 0.0
+    rng = np.random.mtrand.RandomState(42)
+    X = np.abs(rng.randn(48, 5))
+    max_iter = 1
+    init = 'nndsvda'  # FIXME : should be removed in 1.1
+    nmf = NMF(5, solver='mu', init=init, random_state=0,
+              max_iter=max_iter,)
+    mbnmf = MiniBatchNMF(5, solver='mu', init=init, random_state=0,
+                         max_iter=max_iter,
+                         batch_size=X.shape[0], forget_factor=0.0)
+    W = nmf.fit_transform(X)
+    mbW = mbnmf.fit_transform(X)
+    assert_array_almost_equal(W, mbW)
+
+
+@pytest.mark.parametrize('batch_size', [24, 32, 48])
+def test_nmf_close_minibatch_nmf(batch_size):
+    # Test that the decomposition with standard and minibatch nmf
+    # gives close results
+    rng = np.random.mtrand.RandomState(42)
+    X = np.abs(rng.randn(48, 5))
+    max_iter = 5000
+    solver = 'mu'
+    beta_loss = 'kullback-leibler'
+    init = 'nndsvda'  # FIXME : should be removed in 1.1
+    nmf = NMF(5, solver=solver, init=init, random_state=0,
+              max_iter=max_iter, beta_loss=beta_loss)
+    mbnmf = MiniBatchNMF(5, solver=solver, init=init, random_state=0,
+                         max_iter=max_iter, batch_size=batch_size,
+                         beta_loss=beta_loss)
+    W = nmf.fit_transform(X)
+    mbW = mbnmf.fit_transform(X)
+    assert_array_almost_equal(W, mbW, decimal=1)
+
+
+def test_minibatch_nmf_partial_fit():
+    rng = np.random.mtrand.RandomState(42)
+    X = np.abs(rng.randn(48, 5))
+    mbnmf1 = MiniBatchNMF(5, solver='mu', init='nndsvdar', random_state=0,
+                          max_iter=200, batch_size=24)
+    mbnmf2 = MiniBatchNMF(5, solver='mu', init='nndsvdar', random_state=0,
+                          max_iter=1, batch_size=24)
+
+    mbnmf1.fit(X)
+    for i in range(mbnmf1.n_iter_):
+        mbnmf2.partial_fit(X)
+
+    assert mbnmf1.n_iter_ == mbnmf2.n_iter_
+    assert_array_almost_equal(mbnmf1.components_, mbnmf2.components_,
+                              decimal=0)
```

Comment:
with decimal=0 the test doesn't check anything. It's expected that with the current implementation both ways are not equivalent. See my comment above about partial_fit. Making the change suggested above should make this test pass (setting the appropriate params).